### ISSUE TITLE
TemporalTime and TemporalDateTime do not save correctly in DataStore

### ISSUE THREAD

**Issue by Jordan-Nelson** [aws-amplify/amplify-flutter#817]
**Describe the bug**
When using `TemporalTime` or `TemporalDateTime`, the data saved to DataStore does not match the data retrieved from DataStore. The queried value is swapping milliseconds, microseconds, and nanoseconds. This can be observed in the screenshot below, which shows a list of 5 models prior to saving them, and then a list of 5 models after saving/querying.

<img width="772" alt="Screen Shot 2021-08-18 at 1 59 33 PM" src="https://user-images.githubusercontent.com/20613561/129949633-27e2c080-179d-4e92-88c3-ede6bdcfd385.png">

**To Reproduce**
1. Create a schema such as the one below

```graphql
type DateTimeTypeModel @model {
  id: ID!
  dateTimeType: AWSDateTime
}
```
2. Create a new model and save it
```dart
  var now = DateTime.now();
  var model = DateTimeTypeModel(dateTimeType: TemporalDateTime(now));
  await Amplify.DataStore.save(model);
```
3. Query the model
```dart
var queriedModel = await Amplify.DataStore.query(DateTimeTypeModel.classType);
```
4. Compare the models and observe the difference
```dart
print(model == queriedModel);
```

**Expected behavior**
The queried model should equal the saved model

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[] iOS

**NOTE**
Prior to this issue being marked as resolved, the DataStore integration tests that are skipped due to this issue should be updated. 

See: https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_datastore/example/integration_test/model_type_test.dart

**Comment by b-cancel** [comment]
This is a bug report I just filed that includes a couple of bugs I found and the workarounds to each
after reading your bug report I think it could help
https://github.com/aws-amplify/amplify-flutter/issues/822

**Comment by HuiSF** [comment]
This issue has been fixed with version [`0.2.7`](https://github.com/aws-amplify/amplify-flutter/releases/tag/v0.2.7).
